XpressAI/xpipes
# ✨Dropdown menu for run type options (E.g. Spark Submit)

# Description

Added run type option for Spark Submit and run without compiling.

## Pull Request Type


## Type of Change


# Tests

1. Make sure the default **'Run'** works perfectly.
2. Change the python script of its xpipes and then click '**Run w/o compile**'. It should run the changes in the script.
   1. Make sure the python script already exist as this option will run its `.py` not `.xpipes`.
3. Run the simple '**Spark submit**'. 
   1. Make sure your local terminal can run spark submit.
   2. Delete the value of `ADD_ARGS` in config file first.
4. Add any additional argument, `ADD_ARGS` in config file to do more complicated spark job.
   1. Again, make sure your local terminal can run the added arguments of spark submit.
5. Open multiple xpipes then choose the run type. Does it run the correct run type for each xpipes?
   - The run type will save its state for every opened xpipes.  
   - E.g. When one xpipes choose '**Spark Submit**' option, the other opened xpipes will also change its run type to '**Spark Submit**'
   - A new opened xpipes will have the default '**Run**' option.
6. (Optional) Run the '**Spark submit**' in Linux.  


## Tested on?


# Notes

1. Please remove the `ADD_ARGS` to run a simple spark-submit.
2. `ADD_ARGS` in `xai-components/.xpipes/config.ini` is **empty**
3. The `spark-submit` has been tested on Windows using **GitBash**. Currently, I'm unable to run `spark-submit` in **PowerShell**. If you're able to, please pm me how you did it. Thanks.
